lib/plugins/aws/package/compile/events/sqs/index.js

```diff
@@ -95,11 +96,18 @@ class AwsCompileSQSEvents {
               }
             `;
 
+            const sqsTemplateObject = JSON.parse(sqsTemplate);
+
+            if (typeof event.sqs.maximumBatchingWindow !== 'undefined') {
+              sqsTemplateObject.Properties.MaximumBatchingWindowInSeconds =
+                event.sqs.maximumBatchingWindow;
+            }
+
```

Comment:
Let's maybe convert above notation so we directly create `sqsTemplateObject` (creating JSON string, is some old controversial convention used sometimes in a Framework, and in general we're not practising it new code).

Having that configuration would be  simply as;

```javascript
const sqsTemplate = {
  Type: 'AWS::Lambda::EventSourceMapping',
  ...
  Properties: {
    BatchSize,
    ...
  }
}
```

Having that we can include `MaximumBatchingWindowInSeconds` in above construct with a form:

```javascript
  MaximumBatchingWindowInSeconds: event.sqs.maximumBatchingWindow != null ? event.sqs.maximumBatchingWindow : undefined
```

1. `!= null`  checks wether value is neither `null` nor `undefined` (it's the values for which we should assume that property is not set)
2. Assigning `undefined` is like omitting property, as `undefined` properties do not get serialized to JSON.

Reply:
Thank you so much for the help! I've updated it: https://github.com/serverless/serverless/commit/c9da466875da95584c754b991ec1646998f10018